WeScan/Scan/CaptureSessionManager.swift

```diff
@@ -122,6 +130,25 @@ final class CaptureSessionManager: NSObject, AVCaptureVideoDataOutputSampleBuffe
     }
     
     internal func capturePhoto() {
+        // before capture photo, judge `photoOutput` connection input port have valid video media type, if not report a error
+        var captureConnection: AVCaptureConnection? = nil
```

Comment:
This could be simplified this way:
```swift
let captureConnection = photoOutput.connections.first { (connection) -> Bool in
            return connection.inputPorts.first(where: { (port) -> Bool in
                port.mediaType == .video
            }) != nil
        }
```

Reply:
your code is better, but you sure always is `first` object work?  i don't sure so i use for loop.